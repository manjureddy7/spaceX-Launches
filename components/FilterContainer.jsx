import React from "react";
import YearContainer from "./Filters/YearContainer";
import LaunchContainer from "./Filters/LaunchContainer";
import LandingContainer from "./Filters/LandingContainer";
import styles from "../styles/FilterContainer.module.css";

const FilterContainer = () => {
  return (
    <div className={styles.RootContainer} data-testid="FilterContainer">
      <h2>Filters</h2>
      <YearContainer />
      <LaunchContainer />
      <LandingContainer />
    </div>
  );
};

export default FilterContainer;
