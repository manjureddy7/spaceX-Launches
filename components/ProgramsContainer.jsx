import React, { useState, useEffect } from "react";
import Program from "./Programs/Program";
import styles from "../styles/ProgramContainer.module.css";
import { useProgramContext } from "../context/ProgramContext";
import { useRouter } from "next/router";
import getLaunchPrograms from "./data/LaunchPrograms";

export const ProgramsContainer = () => {
  const router = useRouter();
  const [loading, setLoading] = useState(true);
  const [launchPrograms, setLaunchPrograms] = useState([]);
  // const { programsDetails: launchPrograms } = useProgramContext();
  const { landing, launch, year } = router.query;

  useEffect(async () => {
    setLoading(true);
    const response = await getLaunchPrograms(landing, launch, year);
    setLaunchPrograms(response);
    setLoading(false);
  }, [landing, launch, year]);

  return (
    <div className={styles.RootContainer}>
      {loading ? (
        <h1>Loading...</h1>
      ) : launchPrograms.length > 0 ? (
        launchPrograms.map((program) => {
          return (
            <div
              className={styles.IndividualProgram}
              key={program.flight_number}
            >
              <Program
                name={program.mission_name}
                id={program.flight_number}
                missionIds={program.mission_id}
                launchYear={program.launch_year}
                successfulLaunch={program.launch_success}
                successfulLanding={program.rocket.first_stage.cores}
                image={program.links.flickr_images}
              />
            </div>
          );
        })
      ) : (
        <div className={styles.ErrorMessage}>
          No Programs found for the applied search
        </div>
      )}
    </div>
  );
};

export default ProgramsContainer;
