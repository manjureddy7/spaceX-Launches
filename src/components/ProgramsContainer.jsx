import React, { useState, useEffect } from "react";
import Program from "./Programs/Program";
import styles from "../../styles/ProgramContainer.module.css";
import { useRouter } from "next/router";
import getLaunchPrograms from "./api/getLaunchPrograms";

export const ProgramsContainer = () => {
  const router = useRouter();
  const [loading, setLoading] = useState(true);
  const [launchPrograms, setLaunchPrograms] = useState([]);
  const { land_success, launch_success, launch_year } = router.query;

  useEffect(async () => {
    setLoading(true);
    const response = await getLaunchPrograms(
      land_success,
      launch_success,
      launch_year
    );
    setLaunchPrograms(response);
    setLoading(false);
  }, [land_success, launch_success, launch_year]);

  return (
    <div>
      {loading ? (
        <h1>Loading SpaceX Launch Programs...</h1>
      ) : launchPrograms.length > 0 ? (
        <div className={styles.RootContainer}>
          {launchPrograms.map((program, index) => {
            return (
              <div className={styles.IndividualProgram} key={index}>
                <Program
                  name={program.mission_name}
                  id={program.flight_number}
                  missionIds={program.mission_id}
                  launchYear={program.launch_year}
                  successfulLaunch={program.launch_success}
                  successfulLanding={program.rocket.first_stage.cores}
                  image={program.links.flickr_images}
                />
              </div>
            );
          })}
        </div>
      ) : (
        <div className={styles.ErrorMessage}>
          No Programs found for the applied search
        </div>
      )}
    </div>
  );
};

export default ProgramsContainer;
