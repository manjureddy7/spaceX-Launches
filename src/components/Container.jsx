import React from "react";
import FilterContainer from "./FilterContainer";
import ProgramsContainer from "./ProgramsContainer";
import styles from "../../styles/Container.module.css";

const Container = () => {
  return (
    <div data-testid="container" className={styles.RootContainer}>
      <div className={styles.LeftContainer}>
        <FilterContainer />
      </div>
      <div className={styles.RightContainer}>
        <ProgramsContainer />
      </div>
    </div>
  );
};

export default Container;
