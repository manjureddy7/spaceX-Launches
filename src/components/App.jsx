import React from "react";
import Container from "./Container";
import styles from "../../styles/App.module.css";

const App = () => {
  return (
    <div className={styles.AppContainer} data-testid="App">
      <h1 className={styles.AppHeader}>SpaceX Launch Programs</h1>
      <Container />
    </div>
  );
};

export default App;
