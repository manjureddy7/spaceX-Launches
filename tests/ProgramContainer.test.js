import React from "react";
import { render } from "@testing-library/react";
import { ProgramsContainer } from "../components/ProgramsContainer";

const useRouter = jest.spyOn(require("next/router"), "useRouter");

describe("ProgramsContainer", () => {
  it("renders - ProgramsContainer", () => {
    useRouter.mockImplementationOnce(() => ({
      query: {},
    }));
    const { container } = render(<ProgramsContainer />);
    expect(container).toMatchSnapshot();
  });
});
