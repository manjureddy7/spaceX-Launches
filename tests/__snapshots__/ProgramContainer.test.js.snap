// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgramsContainer renders - ProgramsContainer 1`] = `
<div>
  <div
    class="RootContainer"
  >
    <h1>
      Loading...
    </h1>
  </div>
</div>
`;
